From 79edb4779f17f18a1b197b5930272c5af5d55593 Mon Sep 17 00:00:00 2001
From: kelson8 <kelson@kelsoncraft.net>
Date: Sun, 1 Mar 2026 00:19:08 -0500
Subject: [PATCH 54/58] refactor: Add documentation

* Added documentation to Fire.cpp, Game.cpp, and PlayerInfo.cpp.

From this commit:
https://git.internal.kelsoncraft.net/kelson8/re3/commit/482208651ecde5259bfc2749d5336cffc4ce4bfb
---
 src/core/Fire.cpp       |  93 +++++++++++++++++++++++++++++++++++
 src/core/Game.cpp       |  29 +++++++++++
 src/core/PlayerInfo.cpp | 105 ++++++++++++++++++++++++++++++++++++++++
 3 files changed, 227 insertions(+)

diff --git a/src/core/Fire.cpp b/src/core/Fire.cpp
index 57315b15..49db5ccf 100644
--- a/src/core/Fire.cpp
+++ b/src/core/Fire.cpp
@@ -38,6 +38,9 @@ CFire::CFire()
 
 CFire::~CFire() {}
 
+/// <summary>
+/// Process the fire around the player, vehicles and everywhere else.
+/// </summary>
 void
 CFire::ProcessFire(void)
 {
@@ -159,6 +162,10 @@ CFire::ProcessFire(void)
 	}
 }
 
+
+/// <summary>
+/// Report a fire in this area.
+/// </summary>
 void
 CFire::ReportThisFire(void)
 {
@@ -166,6 +173,9 @@ CFire::ReportThisFire(void)
 	CEventList::RegisterEvent(EVENT_FIRE, m_vecPos, 1000);
 }
 
+/// <summary>
+/// Extinguish a fire.
+/// </summary>
 void
 CFire::Extinguish(void)
 {
@@ -200,6 +210,14 @@ CFire::Extinguish(void)
 	}
 }
 
+/// <summary>
+/// Start a fire.
+/// </summary>
+/// <param name="entityOnFire">The entity to be on fire</param>
+/// <param name="fleeFrom"></param>
+/// <param name="strength">Strength of the fire</param>
+/// TODO Figure out what below is.
+/// <param name="propagation">Possibly how far the fire spreads.</param>
 void
 CFireManager::StartFire(CVector pos, float size, uint8 propagation)
 {
@@ -222,6 +240,15 @@ CFireManager::StartFire(CVector pos, float size, uint8 propagation)
 	}
 }
 
+/// <summary>
+/// Start a fire, this one doesn't seem to be in use
+/// </summary>
+/// <param name="entityOnFire">The entity to be on fire</param>
+/// <param name="fleeFrom"></param>
+/// <param name="strength">Strength of the fire</param>
+/// TODO Figure out what below is.
+/// <param name="propagation">Possibly how far the fire spreads.</param>
+/// <returns>The fire that was just started possibly.</returns>
 CFire *
 CFireManager::StartFire(CEntity *entityOnFire, CEntity *fleeFrom, float strength, uint8 propagation)
 {
@@ -309,6 +336,10 @@ CFireManager::StartFire(CEntity *entityOnFire, CEntity *fleeFrom, float strength
 	return fire;
 }
 
+
+/// <summary>
+/// Updates the fire manager, and process ongoing fires.
+/// </summary>
 void
 CFireManager::Update(void)
 {
@@ -318,6 +349,12 @@ CFireManager::Update(void)
 	}
 }
 
+/// <summary>
+/// Find the closest fire with the vecPos from the pDistance.
+/// </summary>
+/// <param name="vecPos">The location to search for fires in.</param>
+/// <param name="pDistance">The distance to search for fires.</param>
+/// <returns></returns>
 CFire* CFireManager::FindNearestFire(CVector vecPos, float *pDistance)
 {
 	int fireId = -1;
@@ -340,6 +377,14 @@ CFire* CFireManager::FindNearestFire(CVector vecPos, float *pDistance)
 	return nil;
 }
 
+/// <summary>
+/// TODO Figure out what this is doing.
+/// Unknown as to what this does
+/// </summary>
+/// <param name="coords"></param>
+/// <param name="minRange"></param>
+/// <param name="maxRange"></param>
+/// <returns></returns>
 CFire *
 CFireManager::FindFurthestFire_NeverMindFireMen(CVector coords, float minRange, float maxRange)
 {
@@ -362,6 +407,11 @@ CFireManager::FindFurthestFire_NeverMindFireMen(CVector coords, float minRange,
 		return &m_aFires[furthestFire];
 }
 
+/// <summary>
+/// TODO Figure out what this is doing.
+/// Get the next free fire in the script possibly, unknown what this actually does
+/// </summary>
+/// <returns></returns>
 CFire *
 CFireManager::GetNextFreeFire(void)
 {
@@ -372,12 +422,22 @@ CFireManager::GetNextFreeFire(void)
 	return nil;
 }
 
+/// <summary>
+/// 
+/// </summary>
+/// <param name=""></param>
+/// <returns>The total amount of active fires</returns>
 uint32
 CFireManager::GetTotalActiveFires(void) const
 {
 	return m_nTotalFires;
 }
 
+/// <summary>
+/// Extinguish a certain point of fire.
+/// </summary>
+/// <param name="point">CVector of where the point is.</param>
+/// <param name="range">The range of the fire</param>
 void
 CFireManager::ExtinguishPoint(CVector point, float range)
 {
@@ -389,6 +449,13 @@ CFireManager::ExtinguishPoint(CVector point, float range)
 	}
 }
 
+/// <summary>
+/// TODO Figure out why this is a boolean
+/// Extinguish a certain point of fire with water
+/// </summary>
+/// <param name="point">CVector of where the point is.</param>
+/// <param name="range">The range of the fire</param>
+/// <returns>Unknown</returns>
 bool
 CFireManager::ExtinguishPointWithWater(CVector point, float range)
 {
@@ -417,6 +484,15 @@ CFireManager::ExtinguishPointWithWater(CVector point, float range)
 	return true;
 }
 
+/// <summary>
+/// Start a fire in an area
+/// </summary>
+/// <param name="pos">Vector3 coords to location</param>
+/// <param name="target">Entity to target</param>
+/// <param name="strength">Strength of the fire</param>
+/// TODO Figure out what below is.
+/// <param name="propagation">Possibly how far the fire spreads.</param>
+/// <returns></returns>
 int32
 CFireManager::StartScriptFire(const CVector &pos, CEntity *target, float strength, uint8 propagation)
 {
@@ -470,12 +546,20 @@ CFireManager::StartScriptFire(const CVector &pos, CEntity *target, float strengt
 	return fire - m_aFires;
 }
 
+/// <summary>
+/// Check if the script fire is extinguished or not.
+/// </summary>
+/// <param name="index">The fire to check for</param>
+/// <returns>If the script fire is extinguished or not</returns>
 bool
 CFireManager::IsScriptFireExtinguish(int16 index)
 {
 	return !m_aFires[index].m_bIsOngoing;
 }
 
+/// <summary>
+/// Remove all script fires
+/// </summary>
 void
 CFireManager::RemoveAllScriptFires(void)
 {
@@ -486,6 +570,10 @@ CFireManager::RemoveAllScriptFires(void)
 	}
 }
 
+/// <summary>
+/// Remove the script fire
+/// </summary>
+/// <param name="index">The fire to remove</param>
 void
 CFireManager::RemoveScriptFire(int16 index)
 {
@@ -493,6 +581,11 @@ CFireManager::RemoveScriptFire(int16 index)
 	m_aFires[index].m_bIsScriptFire = false;
 }
 
+/// <summary>
+/// Sets the audio to a script fire
+/// </summary>
+/// <param name="index">The fire to set audio for</param>
+/// <param name="state">Enabled or disabled.</param>
 void
 CFireManager::SetScriptFireAudio(int16 index, bool state)
 {
diff --git a/src/core/Game.cpp b/src/core/Game.cpp
index bca8fbe3..ade9d9d1 100644
--- a/src/core/Game.cpp
+++ b/src/core/Game.cpp
@@ -211,6 +211,10 @@ void ReplaceAtomicPipeCallback();
 #endif // PS2_ALPHA_TEST
 #endif // !LIBRW
 
+/// <summary>
+/// Initialize the render ware engine.
+/// </summary>
+/// <returns></returns>
 bool
 CGame::InitialiseRenderWare(void)
 {
@@ -303,6 +307,9 @@ CGame::InitialiseRenderWare(void)
 	return (true);
 }
 
+/// <summary>
+/// Shuts down the render ware engine.
+/// </summary>
 void CGame::ShutdownRenderWare(void)
 {
 #ifdef SCREEN_DROPLETS
@@ -392,6 +399,13 @@ CGame::FinalShutdown(void)
 	CdStreamShutdown();
 }
 
+
+/// <summary>
+/// Looks like the main loading for the game.
+/// Hmm I could play with this and stop certain features from being enabled.
+/// </summary>
+/// <param name="datFile"></param>
+/// <returns>TODO Possibly if the game is loaded?</returns>
 bool CGame::Initialise(const char* datFile)
 {
 	ResetLoadingScreenBar();
@@ -881,6 +895,9 @@ void CGame::InitialiseWhenRestarting(void)
 #endif
 }
 
+/// <summary>
+/// Process everything for the game such as events.
+/// </summary>
 void CGame::Process(void)
 {
 #if defined(RW_D3D9) && defined(_WIN32)
@@ -1381,6 +1398,10 @@ void CGame::ProcessTidyUpMemory(void)
 #endif
 }
 
+/// <summary>
+/// TODO Look into using something like this and making it to where the game doesn't pause when minimized.
+/// Initialize the game after focus loss.
+/// </summary>
 void
 CGame::InitAfterFocusLoss()
 {
@@ -1391,6 +1412,10 @@ CGame::InitAfterFocusLoss()
 		FrontEndMenuManager.m_bStartUpFrontEndRequested = true;
 }
 
+/// <summary>
+/// Check if the player can see water from current area.
+/// </summary>
+/// <returns>The current area</returns>
 bool
 CGame::CanSeeWaterFromCurrArea(void)
 {
@@ -1398,6 +1423,10 @@ CGame::CanSeeWaterFromCurrArea(void)
 		|| currArea == AREA_HOTEL;
 }
 
+/// <summary>
+/// Check if the player can see outside from current area.
+/// </summary>
+/// <returns>The current area</returns>
 bool
 CGame::CanSeeOutSideFromCurrArea(void)
 {
diff --git a/src/core/PlayerInfo.cpp b/src/core/PlayerInfo.cpp
index 2f16e7d7..fc1170f1 100644
--- a/src/core/PlayerInfo.cpp
+++ b/src/core/PlayerInfo.cpp
@@ -42,6 +42,9 @@ CVector lastPlayerPos;
 bool CPlayerInfo::vehUpsidedownExplode = true;
 #endif
 
+/// <summary>
+/// Resets the player stats back to normal.
+/// </summary>
 void
 CPlayerInfo::Clear(void)
 {
@@ -102,6 +105,9 @@ CPlayerInfo::Clear(void)
 	m_nBustedAudioStatus = BUSTEDAUDIO_NONE;
 }
 
+/// <summary>
+/// Process the player info, pretty much everything to do with the player, updating the stats and everything.
+/// </summary>
 void
 CPlayerInfo::Process(void)
 {
@@ -568,6 +574,11 @@ CPlayerInfo::IsPlayerInRemoteMode()
 	return m_pRemoteVehicle || m_bInRemoteMode;
 }
 
+/// <summary>
+/// Saves the player save info.
+/// </summary>
+/// <param name="buf"></param>
+/// <param name="size"></param>
 void
 CPlayerInfo::SavePlayerInfo(uint8 *buf, uint32 *size)
 {
@@ -597,6 +608,12 @@ CPlayerInfo::SavePlayerInfo(uint8 *buf, uint32 *size)
 #undef CopyToBuf
 }
 
+ 	
+/// <summary>
+/// Loads the player save info.
+/// </summary>
+/// <param name="buf"></param>
+/// <param name="size"></param>
 void
 CPlayerInfo::LoadPlayerInfo(uint8 *buf, uint32 size)
 {
@@ -623,6 +640,15 @@ CPlayerInfo::LoadPlayerInfo(uint8 *buf, uint32 size)
 #undef CopyFromBuf
 }
 
+/// <summary>
+/// TODO Figure out what exactly this does.
+/// Check the position of the car
+/// </summary>
+/// <param name="carToTest">Car to check position to.</param>
+/// <param name="player">The player.</param>
+/// <param name="carBoundCentrePedDist">Distance to check for</param>
+/// <param name="lastCloseness">Unknown</param>
+/// <param name="closestCarOutput">Unknown</param>
 void
 CPlayerInfo::FindClosestCarSectorList(CPtrList& carList, CPed* ped, float unk1, float unk2, float unk3, float unk4, float* lastCloseness, CVehicle** closestCarOutput)
 {
@@ -672,6 +698,10 @@ CPlayerInfo::EvaluateCarPosition(CEntity *carToTest, CPed *player, float carBoun
 	}
 }
 
+/// <summary>
+/// Get the players current position
+/// </summary>
+/// <returns>The players position</returns>
 const CVector &
 CPlayerInfo::GetPos()
 {
@@ -684,6 +714,10 @@ CPlayerInfo::GetPos()
 	return m_pPed->GetPosition();
 }
 
+/// <summary>
+/// Get the players coords.
+/// </summary>
+/// <returns>The X, Y, Z, and possibly rotation.</returns>
 CVector
 FindPlayerCoors(void)
 {
@@ -698,6 +732,10 @@ FindPlayerCoors(void)
 		return ped->GetPosition();
 }
 
+/// <summary>
+/// Get the players speed
+/// </summary>
+/// <returns>The current move speed</returns>
 const CVector &
 FindPlayerSpeed(void)
 {
@@ -713,6 +751,10 @@ FindPlayerSpeed(void)
 		return ped->m_vecMoveSpeed;
 }
 
+/// <summary>
+/// Get the players current vehicle
+/// </summary>
+/// <returns>The vehicle the player is in, or nil if in no vehicle.</returns>
 CVehicle *
 FindPlayerVehicle(void)
 {
@@ -721,6 +763,10 @@ FindPlayerVehicle(void)
 	return nil;
 }
 
+/// <summary>
+/// Find the player entity (Possibly the entity id or handle.)
+/// </summary>
+/// <returns>The ped of the player</returns>
 CEntity *
 FindPlayerEntity(void)
 {
@@ -731,6 +777,11 @@ FindPlayerEntity(void)
 		return ped;
 }
 
+/// <summary>
+/// Check if the vehicle is a train
+/// </summary>
+/// <param name=""></param>
+/// <returns>True if the vehicle is a train, otherwise nil.</returns>
 CVehicle *
 FindPlayerTrain(void)
 {
@@ -740,12 +791,21 @@ FindPlayerTrain(void)
 		return nil;
 }
 
+/// <summary>
+/// Get the current player.
+/// </summary>
+/// <returns>The current player</returns>
 CPlayerPed *
 FindPlayerPed(void)
 {
 	return CWorld::Players[CWorld::PlayerInFocus].m_pPed;
 }
 
+/// <summary>
+/// Find the center of the world for the player.
+/// </summary>
+/// <param name="player">The player.</param>
+/// <returns>The center of the world.</returns>
 const CVector &
 FindPlayerCentreOfWorld(int32 player)
 {
@@ -758,6 +818,11 @@ FindPlayerCentreOfWorld(int32 player)
 	return CWorld::Players[player].m_pPed->GetPosition();
 }
 
+/// <summary>
+/// TODO Figure out what this does.
+/// </summary>
+/// <param name=""></param>
+/// <returns></returns>
 const CVector &
 FindPlayerCentreOfWorld_NoSniperShift(void)
 {
@@ -771,6 +836,10 @@ FindPlayerCentreOfWorld_NoSniperShift(void)
 	return FindPlayerPed()->GetPosition();
 }
 
+// <summary>
+/// Get the players heading.
+/// </summary>
+/// <returns>The heading of the player.</returns>
 float
 FindPlayerHeading(void)
 {
@@ -780,18 +849,30 @@ FindPlayerHeading(void)
 	return FindPlayerPed()->GetForward().Heading();
 }
 
+/// <summary>
+/// Is the player restarting after dying.
+/// </summary>
+/// <returns>If the player is restarting after dying</returns>
 bool
 CPlayerInfo::IsRestartingAfterDeath()
 {
 	return m_WBState == WBSTATE_WASTED;
 }
 
+/// <summary>
+/// Is the player restarting after being busted.
+/// </summary>
+/// <returns>If the player is restarting after being busted</returns>
 bool
 CPlayerInfo::IsRestartingAfterArrest()
 {
 	return m_WBState == WBSTATE_BUSTED;
 }
 
+/// <summary>
+/// Kills the player, and increase the stat
+/// TODO Does this work?
+/// </summary>
 void
 CPlayerInfo::KillPlayer()
 {
@@ -804,6 +885,9 @@ CPlayerInfo::KillPlayer()
 	CStats::TimesDied++;
 }
 
+/// <summary>
+/// Arrest/bust the player.
+/// </summary>
 void
 CPlayerInfo::ArrestPlayer()
 {
@@ -817,6 +901,9 @@ CPlayerInfo::ArrestPlayer()
 	CStats::TimesArrested++;
 }
 
+/// <summary>
+/// Unknown
+/// </summary>
 void
 CPlayerInfo::PlayerFailedCriticalMission()
 {
@@ -838,6 +925,10 @@ CPlayerInfo::CancelPlayerEnteringCars(CVehicle *car)
 		m_pPed->ClearObjective();
 }
 
+/// <summary>
+/// Gives the player invincibility and makes it to where they are ignored, also disables cops.
+/// </summary>
+/// <param name="toggle"></param>
 void
 CPlayerInfo::MakePlayerSafe(bool toggle)
 {
@@ -877,6 +968,10 @@ CPlayerInfo::MakePlayerSafe(bool toggle)
 	}
 }
 
+/// <summary>
+/// Blow up the rc car.
+/// </summary>
+/// <param name="actually>Unknown param.</param>
 void
 CPlayerInfo::BlowUpRCBuggy(bool actually)
 {
@@ -889,6 +984,10 @@ CPlayerInfo::BlowUpRCBuggy(bool actually)
 }
 
 #ifdef GTA_PC
+/// <summary>
+/// Sets the players skin.
+/// </summary>
+/// <param name="skin">The skin to set to.</param>
 void
 CPlayerInfo::SetPlayerSkin(const char *skin)
 {
@@ -896,6 +995,9 @@ CPlayerInfo::SetPlayerSkin(const char *skin)
 	LoadPlayerSkin();
 }
 
+/// <summary>
+/// Loads the players skin.
+/// </summary>
 void
 CPlayerInfo::LoadPlayerSkin()
 {
@@ -904,6 +1006,9 @@ CPlayerInfo::LoadPlayerSkin()
 	m_pSkinTexture = CPlayerSkin::GetSkinTexture(m_aSkinName);
 }
 
+/// <summary>
+/// Deletes the players skin.
+/// </summary>
 void
 CPlayerInfo::DeletePlayerSkin()
 {
-- 
2.49.0.windows.1

